Automattic/woocommerce-payments
# Disable automatic currency switching and switcher widgets on pay_for_order page

Fixes #5031 #6937

#### Changes proposed in this Pull Request

If a manual order is created for any reason, then a customer is taken to a "Pay for Order" page. This page should not allow currency switching in any form since we cannot know if we are converting prices correctly that are set in the order. This PR disables the currency switcher widgets and the automatic currency switching based on geolocation for that page.

* Add new `should_disable_currency_switching` to replace `should_hide_widgets` method. The functionality is the same, however, it is now used to also confirm if the automatic location based currency switching should happen. It has an added check in the method to see if the `pay_for_order` parameter has been passed.
* Instances of `should_hide_widgets` were replaced with `should_disable_currency_switching`, which is a large part of the changes.
* The `render_block_widget` method did not have a check to make sure the widget was not displayed if there was only a single currency, so I've added that.
* Tests were update and/or added as needed.

#### Testing instructions

* Do not check out this branch yet.
* For testing purposes, you will need to add the below filter. Note that this will set all users' geolocation to Canada, so it is assuming your store is not based in Canada and does not have CAD as the default currency.
* Navigate to WooCommerce > Settings > Multi-Currency and:
  * Add CAD as a currency if not already added.
  * Scroll down to _Store settings_ and enable the _Automatically switch customers to their local currency if it has been enabled_ setting.
* The currency switcher widget/block will need to active on your site, this is under Appearance > Widgets for non-block themes. I am not sure how it works for block themes.
* Create a new Customer user under Users > Add new. Take note of the password created.
* Navigate to WooCommerce > Orders, then use the _Add new_ button at the top.
* Create a new order with:
  * A single simple product,
  * The new user as the Customer,
  * The status of Pending Payment.
* Once the order is created, there will be a _Customer payment page_ link near the Status, open that in an incognito window.
* Log in as the new user.
* Confirm that:
  * The currency switcher widget is showing,
  * The banner at the bottom reads something like, "We noticed you're visiting from Canada. We've updated our prices to Canadian dollar..."
* Now check out this branch.
* Refresh the Pay for Order page in your incognito window.
* Confirm that:
  * The currency switcher is no longer showing,
  * The banner at the bottom about updating the currency no longer shows.

Filter for geolocation:
```
	add_filter(
		'woocommerce_geolocate_ip',
		function() {
			return 'CA';
		}
	);
```

-------------------
